Sources/FoundationEssentials/Subprocess/Subprocess+Configuration.swift

```diff
@@ -0,0 +1,615 @@
+        var result = Dictionary<
+            Subprocess.StringOrRawBytes,
+            Subprocess.StringOrRawBytes
+        >(minimumCapacity: self.count)
+        for (key, value) in self {
+            result[.string(key)] = .string(value)
+        }
+        return result
+    }
+}
+
+fileprivate extension Dictionary where Key == Data, Value == Data {
+    func wrapToStringOrRawBytes() -> [Subprocess.StringOrRawBytes : Subprocess.StringOrRawBytes] {
+        var result = Dictionary<
+            Subprocess.StringOrRawBytes,
+            Subprocess.StringOrRawBytes
+        >(minimumCapacity: self.count)
+        for (key, value) in self {
+            result[.rawBytes(key.toArray())] = .rawBytes(value.toArray())
+        }
+        return result
+    }
+}
+
+fileprivate extension Data {
+    func toArray<T>() -> [T] {
+        return self.withUnsafeBytes { ptr in
+            return Array(ptr.bindMemory(to: T.self))
+        }
+    }
+}
+
+// MARK: - ProcessIdentifier
+extension Subprocess {
+    public struct ProcessIdentifier: Sendable, Hashable {
+        let value: pid_t
+
+        internal init(value: pid_t) {
+            self.value = value
+        }
+    }
+}
+
+// MARK: - TerminationStatus
+extension Subprocess {
+    public enum TerminationStatus: Sendable, Hashable {
+        #if canImport(WinSDK)
+        public typealias Code = DWORD
+        #else
+        public typealias Code = CInt
+        #endif
+
+        #if canImport(WinSDK)
+        case stillActive
+        #endif
+
+        case exit(Code)
+        case unhandledException(Code)
+
+        public var isSuccess: Bool {
+            switch self {
+            case .exit(let exitCode):
+                return exitCode == 0
+            case .unhandledException(_):
+                return false
+            }
+        }
+
+        public var isUnhandledException: Bool {
+            switch self {
+            case .exit(_):
+                return false
+            case .unhandledException(_):
+                return true
+            }
+        }
+    }
+}
+
+// MARK: - Internal
+extension Subprocess {
+    internal enum StringOrRawBytes: Sendable, Hashable {
+        case string(String)
+        case rawBytes([CChar])
+
+        // Return value needs to be deallocated manually by callee
+        func createRawBytes() -> UnsafeMutablePointer<CChar> {
+            switch self {
+            case .string(let string):
+                return strdup(string)
+            case .rawBytes(let rawBytes):
+                return strdup(rawBytes)
+            }
+        }
+
+        var stringValue: String? {
+            switch self {
+            case .string(let string):
+                return string
+            case .rawBytes(let rawBytes):
+                return String(validatingUTF8: rawBytes)
+            }
+        }
+
+        var count: Int {
+            switch self {
+            case .string(let string):
+                return string.count
```

Comment:
Are you sure you want the count of the `Character`s here instead of the UTF-8 count or otherwise?